app/services/activitypub/process_account_service.rb

```diff
@@ -197,6 +199,21 @@ def actor_type
     end
   end
 
+  def image_description(key)
+    value = first_of_value(@json[key])
+
+    return if value.nil?
+
+    if value.is_a?(String)
+      value = fetch_resource_without_id_validation(value)
+      return if value.nil?
+    end
+
+    value = first_of_value(value['name']) if value.is_a?(Hash) && value['type'] == 'Image'
+    value = value['summary'] if value.is_a?(Hash)
```

Comment:
Not quite, we can only get description if the value is a Hash.

So, like:

```
  return unless value.is_a(Hash)

description = value['summary'].presence || value['name'].presence
description = description.strip[0...Account::Avatar::MAX_DESCRIPTION_LENGTH] if description.present?
description
```

You may want to refactor this to do:

```
    value = first_of_value(@json[key])

    return if value.nil?

    if value.is_a?(String)
      value = fetch_resource_without_id_validation(value)
      return if value.nil?
    end
```

As a helper method `def image(key)` or something, such that you don't duplicate the "fetch the image object" logic.

Reply:
The description truncation should probably also happen based on the `key` (to negotiate which class has the relevant max length) or do it inline in the setters.